pallets/stateful-storage/src/stateful_child_tree.rs

```diff
@@ -0,0 +1,395 @@
+use core::marker::PhantomData;
+
+use codec::{Decode, Encode};
+use common_primitives::msa::MessageSourceId;
+use frame_support::{
+	storage::{child, child::ChildInfo, ChildTriePrefixIterator},
+	Blake2_128, Blake2_256, Hashable, StorageHasher, Twox128, Twox256,
+};
+use sp_core::{ConstU8, Get};
+use sp_io::hashing::twox_64;
+use sp_std::{fmt::Debug, prelude::*};
+
+pub trait MultipartKeyStorageHasher: StorageHasher {
```

Comment:
pretty cool how you create this trait with a default and did implementations below.